mvc/src/main/webapp/WEB-INF/jsp/mushrooms/edit.jsp

```diff
@@ -0,0 +1,57 @@
+<%--
+  Created by IntelliJ IDEA.
+  User: bencikpeter
+  Date: 16/12/2017
+  Time: 12:35
+  To change this template use File | Settings | File Templates.
+--%>
+<%@ page contentType="text/html;charset=UTF-8" pageEncoding="utf-8" trimDirectiveWhitespaces="false" session="true" %>
+<%@ taglib tagdir="/WEB-INF/tags" prefix="my" %>
+<%@ taglib prefix="c" uri="http://java.sun.com/jsp/jstl/core" %>
+<%@ taglib prefix="fmt" uri="http://java.sun.com/jsp/jstl/fmt" %>
+<%@ taglib prefix="s" uri="http://www.springframework.org/tags" %>
+<%@ taglib prefix="form" uri="http://www.springframework.org/tags/form" %>
+
+<my:pagetemplate title="Edit mushroom">
+<jsp:attribute name="body">
+  <c:set var="end" value="mushrooms"/>
+
+    <form:form method="post" action="${pageContext.request.contextPath}/${end}/edit/${mushroomEdit.id}"
+               modelAttribute="mushroomEdit" cssClass="form-horizontal">
+
+        <div class="form-group ${name_error?'has-error':''}">
+            <form:label path="name" cssClass="col-sm-2 control-label">Name</form:label>
+            <div class="col-sm-10">
+                <form:input path="name" cssClass="form-control"/>
+                <form:errors path="name" cssClass="help-block"/>
+            </div>
+        </div>
+
+        <div class="form-group ${type_error?'has-error':''}">
+            <form:label path="type" cssClass="col-sm-2 control-label">Type</form:label>
+            <div class="col-sm-10">
+                <form:select path="type" cssClass="form-control">
+                    <c:forEach items="${mushroomType}" var="c">
+                        <form:option value="${c}">${c}</form:option>
+                    </c:forEach>
+                </form:select>
+                <form:errors path="type" cssClass="help-block"/>
+            </div>
+        </div>
+       <div class="form-group ${intervalOfOccurrence_error?'has-error':''}"> <!-- TODO: some dropdown instead of text? -->
```

Comment:
some sort of dropdown menu would be good - we'll save us some validation too.

Reply:
Id' suggest a dropdown for 'from' and 'to' month, which would be saved to the DTO object as the correct string (month - month)